packages/components/src/components/TextField/TextField.tsx

```diff
@@ -4,38 +4,95 @@ import React, { forwardRef } from "react";
 import { Typography } from "../Typography";
 import { EDSStyleSheet } from "../../styling";
 import { useStyles } from "../../hooks/useStyles";
+import { Label } from "../Label";
+import { IconName, Icon } from "../Icon";
 
 export type TextFieldProps = {
+    /**
+     * A small label text to add to the input field.
+     */
+    label?: string;
+
+    /**
+     * Secondary small label text to add to the input field.
+     */
+    meta?: string;
+
     /**
      * A unit describing the input value.
      */
     unit?: string;
+
+    /**
+     * A description to add to the input field. Use this when more information around the input field is required.
+     */
+    helperText?: string;
+
+    /**
+     * An icon to add to inside the inputfield on the right side.
+     */
+    inputIcon?: IconName;
+
+    /**
+     * An icon to add to the left of the helper text.
+     */
+    helperIcon?: IconName;
 } & Omit<InputProps, "leftAdornment" | "rightAdornment">;
 
-export const TextField = forwardRef<TextInput, TextFieldProps>(({ unit, ...rest }, ref) => {
-    const styles = useStyles(themeStyles);
-    return (
-        <Input
-            ref={ref}
-            rightAdornments={
-                unit && (
-                    <View style={styles.unit}>
-                        <Typography variant="label" color="textTertiary">
-                            {unit}
-                        </Typography>
+export const TextField = forwardRef<TextInput, TextFieldProps>(
+    ({ unit, helperText, label, meta, helperIcon, inputIcon, ...rest }, ref) => {
+        const styles = useStyles(themeStyles, { variant: rest.variant });
+        return (
+            <View style={{ flexGrow: 1 }}>
```

Comment:
Added flexGrow in order to make it more versatile when used in other projects and in a flex grid 🤠